### DISCUSSION TITLE
ARKit vs. ARCore using from C#/.NET

### DISCUSSION THREAD

**Discussion by llfab** [StereoKit/StereoKit#1099] [Q&A]
We had done an Android based app for AR using Java and ARCore. We may want to do a rewrite using C#/.NET and are free to select either iOS or Android as the target platform. So, we wonder what your experiences are using either ARCore (Android) or ARKit (iOS) from C#.

**Comment by maluoi** [comment]
I'm afraid I don't have terribly much experience with ARCore _or_ ARKit. I've spent a bit more time with ARCore simply because Android has generally been nicer to dev for, but not really enough to develop an opinion one way or the other.

I have generally gotten the impression that ARKit may be more advanced than ARCore feature-wise.

If you're interested, there is a small demo of ARCore with StereoKit over here: https://github.com/maluoi/SKARCore

**Reply by llfab** [reply]
Thanks. as far as I understand creating a SK experience on a tablet is not within the supported devices, right?

**Reply by maluoi** [reply]
StereoKit's simulator runs fine on Android, but has no core support for an ARCore-like experience. The linked SK+ARCore demo is a top level implementation, and only includes device motion and the "passthrough" visuals. This doesn't include fancier features or even input yet. I'll probably be hooking up input to this sometime in the next month or two.

But yeah, if you're looking for a fully featured ARCore-like experience with SK, that's the closest you'll get today.

**Comment by rayperkins** [comment]
@llfab, I'd put together a small learning project end of 2023 that used dotnet 7 & the Android ARCore, we were exploring IoT and Cloud Anchors.

The only technical challenge was to generate up to date bindings as the nuget versions are very old. ARCore lets you pick whatever rendering tech and so would work nicely with SK as per the demo linked by @maluoi.

Here's the binding project I made for ARCore 1.37 [Google.AR.Core.zip](https://github.com/user-attachments/files/16998671/Google.AR.Core.zip)
